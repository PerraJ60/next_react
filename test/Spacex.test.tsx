import TestRenderer from 'react-test-renderer';
import { render, cleanup } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import '@testing-library/jest-dom/extend-expect';
import Spacex, { launchesQuery } from '../pages/Spacex';

afterEach(cleanup);

const mocks = [
  {
    request: {
      query: launchesQuery,
      variables: {},
    },
    result: {
      data: [],
    },
  },
];

it('renders without error', () => {
  const component = TestRenderer.create(
    <MockedProvider mocks={mocks} addTypename={false}>
      <Spacex launches={mocks} />
    </MockedProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
