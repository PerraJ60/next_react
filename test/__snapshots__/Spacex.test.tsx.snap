// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without error 1`] = `
<div>
  <nav>
    <ul
      className="header"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/Products"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Products
        </a>
      </li>
      <li>
        <a
          href="/Continents"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Continents
        </a>
      </li>
      <li>
        <a
          href="/Spacex"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Space travels
        </a>
      </li>
      <li>
        <a
          href="/About"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          CV
        </a>
      </li>
      <li>
        <a
          href="/Contact"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Contact
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="container"
  >
    <div
      className="main"
    >
      <h1
        className="title"
      >
        SpaceX Launches
      </h1>
      <p
        className="description"
      >
        <a
          href="https://www.spacex.com/launches/"
        >
          Latest launches
        </a>
         from SpaceX
      </p>
      <div
        className="grid"
      >
        <a
          className="card"
        >
          <h3 />
          <p>
            <strong>
              Launch Date:
            </strong>
             
            Invalid Date
          </p>
        </a>
      </div>
    </div>
  </div>
</div>
`;
